src/routes/quickstarts/cypress/setup.mdx

```diff
@@ -0,0 +1,180 @@
+
+```js
+// src/server.js
+import { Server, Model } from "miragejs"
+
+export function makeServer({ environment = "development" } = {}) {
+  let server = new Server({
+    environment,
+
+    models: {
+      user: Model,
+    },
+
+    seeds(server) {
+      server.create("user", { name: "Bob" })
+      server.create("user", { name: "Alice" })
+    },
+
+    routes() {
+      this.namespace = "api"
+
+      this.get("/users", schema => {
+        return schema.users.all()
+      })
+    },
+  })
+
+  return server
+}
+```
+
+## Step 3: Add the Mirage Cypress configuration
+
+Add the following Mirage bootstrapping code and command to your `cypress/support/index.js` file:
+
+```js
+// cypress/support/index.js
+Cypress.on("window:before:load", win => {
+  win.handleFromCypress = function(request) {
+    return fetch(request.url, {
+      method: request.method,
+      body: request.requestBody,
+    }).then(res => {
+      if (res.headers.map["content-type"] === "application/json") {
+        return res.json()
+      } else {
+        return ""
+      }
+    })
+  }
+})
+```
+
+## Step 4: Start Mirage when Cypress is enabled
+
+In your app's bootstrapping file, start Mirage if `window.Cypress` is true.
+
+> Create React App users, this code goes in `src/index.js`
+
+> Vue CLI users, this code goes in `src/main.js`.
+
+```js
+import { Server } from "miragejs"
+
+if (window.Cypress) {
+  let cyInterceptor = new Server({
+    routes() {
+      let methods = ["get", "put", "patch", "post", "delete"]
+      methods.forEach(method => {
+        this[method]("/*", (schema, request) => {
+          return window.handleFromCypress(request)
+        })
+      })
+    },
+  })
+  cyInterceptor.logging = false
+}
+```
+
+## Step 5: Pick a component to test
+
+Here's the component we'll be testing.
+
+<Tabs>
+  <Tab name="React">
+    <Snippet name="cypress-quickstart-react-component" />
+  </Tab>
+  <Tab name="Vue">
+    <Snippet name="cypress-quickstart-vue-component" />
+  </Tab>
+</Tabs>
+
+## Step 6: Write tests using your Mirage server
+
+Create a new `tests/e2e/specs/App.spec.js` file and use this test to seed Mirage with different data scenarios. Then assert against the state of your UI.
+
+```js
+import { makeServer } from "../../src/server"
+
+let server
+
+beforeEach(() => {
+  server = makeServer({ environment: "test" })
+})
+
+afterEach(() => {
+  server.shutdown()
+})
+
+it("shows the users from our server", () => {
+  server.create("user", { id: 1, name: "Luke" })
+  server.create("user", { id: 2, name: "Leia" })
+
+  cy.visit("/")
+
+  cy.get('[data-testid="user-1"]').contains("Luke")
+  cy.get('[data-testid="user-2"]').contains("Leia")
+})
+
+it("shows a message if there are no users", () => {
+  // Don't create any users
+
+  cy.visit("/")
+
+  cy.get('[data-testid="no-users"]').should("be.visible")
+})
+```
+
+> In the `test` environment, Mirage doesn't load its database `seeds`, so that the server starts out empty for each test run.
```

Comment:
Small tweak. And let's drop the `>` backquote and just be a bit more verbose.

Note that we pass in `environment: test` option into our `makeServer` function, so that Mirage doesn't load its database `seeds`. That way, the server starts out empty for each test run, and in the beginning of our tests we can use `server.create` to set up our data scenario.

The `test` environment also disables logging and latency, so that by default your CI test logs will be clean + your tests will be fast. You can set `server.logging = true` if you're writing a test and want to see all the network requests logged to the console.

Reply:
> You can set server.logging = true if you're writing a test and want to see all the network requests logged to the console.

Up until this point the guide has been pretty robotic (copy this, paste that). This part struck me as a teaching bit, but I think I don't really need to know about it now?

Are you thinking this is important because usually you'd see network logging in the cypress UI, but now you don't?

Reply:
Good call – yeah I think you're right, if we want to address this (because we think it will be confusing to Cypress users), we should change it to something like "Compared to Cypress mocking, Mirage mocks won't show up in the sidebar. In order to see them, you can... "

Let's leave this out for now and see how it feels/what the feedback is. Can always drop it in maybe at the top soon.